semeio/workflows/misfit_preprocessor/hierarchical_config.py

```diff
@@ -105,18 +106,28 @@ class LinkageConfig(BaseMisfitPreprocessorConfig):
     ] = "euclidean"
 
 
-class HierarchicalConfig(BaseMisfitPreprocessorConfig):
+def _flatten(dictionary):
+    items = []
+    for key, value in dictionary.items():
+        if isinstance(value, collections.MutableMapping):
+            items.extend(_flatten(value).items())
+        else:
+            items.append((key, value))
+    return dict(items)
+
+
+class AbstractClusteringConfig(BaseMisfitPreprocessorConfig):
+    def cluster_args(self):
```

Comment:
Not sure this is the best approach, but the goal is for the configuration to be able to provide the arguments to the clustering function when we add a new function. In the case of `hierarchical` the configuration is split into `linkage` and `fcluster`, but the function we are configuring is not so the return `dict` is flattened. It could be argued that either the function or the configuration should change, but I kind of like it as it sits.

Reply:
Seems like a reasonable approach, maybe

```
def cluster_args(config_object):
    return _flatten(config_object.dict())
```